File: Caladan.Frontend/ViewModels/Block.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Caladan.Frontend.ViewModels
{
    public class Block : Base
    {
        public ulong BlockNumber { get; set; }
        public string Hash { get; set; }
        public string Miner { get; set; }
        public int NumberOfTransactions { get; set; }
        public string ParentHash { get; set; }
        public string Nonce { get; set; }
        public string Sha3Uncles { get; set; }
        public string LogsBloom { get; set; }
        public string TransactionsRoot { get; set; }
        public string StateRoot { get; set; }
        public ulong Difficulty { get; set; }
        public ulong TotalDifficulty { get; set; }
        public string ExtraData { get; set; }
        public ulong Size { get; set; }
        public ulong GasLimit { get; set; }
        public ulong GasUsed { get; set; }
        public ulong Timestamp { get; set; }
        public string FoundOnFormatted { get; set; }
        public string Url
        {
            get
            {
                return $"/block/{BlockNumber}";
            }
        }
        public SimpleTransaction[] Transactions { get; set; }
    }
}

File: Caladan.Frontend/ClientApp/components/block/block.ts
import Vue from 'vue';
import { Component } from 'vue-property-decorator';

interface Block {
    BlockNumber: number;
    Hash: string;
    Miner: string;
    NumberOfTransactions: number;
    ParentHash: string;
    Nonce: string;
    Sha3Uncles: string;
    LogsBloom: string;
    TransactionsRoot: string;
    StateRoot: string;
    Difficulty: number;
    TotalDifficulty: number;
    ExtraData: string;
    Size: number;
    GasLimit: number;
    GasUsed: number;
    Timestamp: number;
    Url: string;
    FoundOnFormatted: string;
    Found: boolean;
    Transactions: Transaction[];
}

interface Transaction {
    TransactionHash: string;
    BlockNumber: number;
    From: string;
    To: string;
    Value: number;
    Symbol: string;
    Url: string;
    ConfirmedOnFormatted: string;
}

@Component
export default class BlockComponent extends Vue {
    block: Block;
    data() {
        return {
            block: null
        }
    }

    mounted() {
        fetch('internalapi/Block/Get?blockNumber=' + this.$route.params.blockNumber)
            .then(response => response.json() as Promise<Block>)
            .then(data => {
                this.block = data;
            });
    }

    subStr(input: string, length: number) {
        return input.substring(0, length) + '...';
    }
}
